provisioning/device/src/x509_registration.ts

```diff
@@ -43,16 +46,15 @@ export class X509Registration implements RegistrationClient {
           provisioningHost: this._provisioningHost,
           idScope: this._idScope
         };
+        /* Codes_SRS_NODE_DPS_X509_REGISTRATION_18_004: [ `register` shall pass the certificate into the `setAuthentication` method on the transport ] */
+        this._transport.setAuthentication(cert);
         /* Codes_SRS_NODE_DPS_X509_REGISTRATION_18_002: [ `register` shall call `registerX509` on the transport object and call it's callback with the result of the transport operation. ] */
-        this._transport.registerX509(request, cert, (err, result, body) => {
+        this._pollingStateMachine.register(request, (err, response) => {
```

Comment:
>response [](start = 58, length = 8)

Historically the convention for callback parameters in the device SDK is the following:
- result is an object that is independent of the transport
- response in the HTTP object we receive from the server (after we make our request -and it's used only the the context of HTTP)
- responseBody if the actual body of the HTTP response

all callback are usually (err: Error, result: SomeType) => void and sometimes (err: Error, result: SomeType, response: IncomingMessage) => void.

I would very much like a pass for consistency in the provisioning SDK to avoid confusion in the future when we go back to these pieces of code to debug them

Reply:
Agree on the need for consistency.  Adding it to my TFS task for later.

---
In reply to: [155607744](https://github.com/Azure/azure-iot-sdk-node/pull/176#discussion_r155607744) [](ancestors = 155607744)